test/verify/testsuite-prepare

```diff
@@ -1,87 +1,66 @@
+import os
+import shutil
+import sys
+import imp
 
-usage()
-{
-    echo "usage: testsuite-prepare [--clean] [--quick] [--verbose]"
-}
+basepath = os.path.join(os.path.dirname(os.path.abspath(__file__)),"..")
+if not basepath in sys.path:
+    sys.path.insert(1, basepath)
+from common import testinfra
+vminstall = imp.load_source('vminstall', os.path.join(basepath,"vm-install"))
+vmdownload = imp.load_source('vmdownload', os.path.join(basepath,"vm-download"))
 
+image_list="ipa openshift fedora-stock"
 
-args=$(getopt -o "h,v,q" -l "help,verbose,quick" -- "$@")
-eval set -- "$args"
-while [ $# -gt 0 ]; do
-    case $1 in
-    -q|--quick)
-        vm_install_opts="$vm_install_opts --quick"
-        ;;
-    -v|--verbose)
-        vm_install_opts="$vm_install_opts --verbose"
-        ;;
-    -h|--help)
-        usage
-        exit 0
-        ;;
-    --)
-        shift
-        break
-        ;;
-    esac
-    shift
-done
+parser = argparse.ArgumentParser(
+        description='Prepare testig environment, download images and build and cinstall cockpit',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+parser.add_argument('-v', '--verbose', action='store_true', help='Display verbose progress details')
+parser.add_argument('-n', '--noinstall', action='store_true', help='Do not install cockpit to base image')
+parser.add_argument('-f', '--force', action='store_true', help='Force update of images')
+parser.add_argument('image', nargs='?', default=testinfra.DEFAULT_IMAGE, help='The image to use')
+args = parser.parse_args()
 
-# The Atomic variants can't build their own packages, so we build in
-# their non-Atomic siblings.  For example, fedora-atomic is build
-# in fedora-23
-#
-if [ "$TEST_OS" = "fedora-atomic" ]; then
-    BUILD_OS="fedora-23"
-elif [ "$TEST_OS" = "rhel-atomic" ]; then
-    BUILD_OS="rhel-7"
-else
-    BUILD_OS="$TEST_OS"
-fi
+test_os = args.image
+build_os = test_os
 
-$BASE/vm-reset
-$BASE/vm-download \
-    $TEST_OS \
-    $BUILD_OS \
-    ipa \
-    openshift \
-    fedora-stock
+if test_os == "fedora-atomic":
+    build_os = "fedora-23"
+elif test_os == "rhel-atomic":
+    build_os = "rhel-7"
 
-install_failed=no
-if ! $BASE/vm-install $vm_install_opts --build-image $BUILD_OS; then
-    install_failed=yes
-fi
+# vm-reset replacement
+shutil.rmtree("%s/tmp/run" % basepath)
+os.makedirs("%s/tmp/run" % basepath)
 
-# Upload the build logs if desired
-if [ -n "${TEST_ATTACHMENTS:-}" ]; then
-    mkdir "$TEST_ATTACHMENTS"/build-results
-    cp -r $BASE/tmp/build-results/*.log "$TEST_ATTACHMENTS"/build-results
-fi
+vmdownload.download_images(set((test_os, build_os)), args.force, [])
+vmdownload.download_images(image_list.split(" "), args.force, [])
 
-if [ $install_failed == "yes" ]; then
-  exit 1
-fi
+if args.noinstall:
+    print "Not installing cockpit to base image"
+else:
+    args.sit = False
+    args.containers = False
+    args.quick = False
+    vminstall.build_and_install(test_os, build_os, args)
```

Comment:
It is little bit dangherous, but from my POV it was cleaner, than creating completely new env in case some values are inherited.
But This part I can completely remove in case I'll rewrite vm-install more. there is on many places hardcodede using args in functions, thats why it is so nonclean.